src/Raven.Server/Documents/DocumentPutAction.cs

```diff
@@ -213,28 +188,65 @@ public DocumentPutAction(DocumentsStorage documentsStorage, DocumentDatabase doc
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private string BuildDocumentKey(DocumentsOperationContext context, string key, Table table, long newEtag, out bool knownNewKey)
+        private ChangeVectorEntry[] BuildChangeVector(DocumentsOperationContext context, string key, Slice lowerKey, long newEtag, ChangeVectorEntry[] changeVector, long? expectedEtag, DocumentFlags flags, TableValueReader oldValue)
```

Comment:
Misleading method name here, you are also resolving the conflict here, and it is hidden and not visible for us.